### PR TITLE
Upgrade grape dependency to include 1.3

### PR DESCRIPTION
**Pull request by sdfreund10**
Response to https://github.com/antek-drzewiecki/wine_bouncer/issues/80
1) Remove the upper bound limit on grape 1.3, to allow the upgrade
2) Patch Gemfile to automagically pull compatible versions of gems
3) Fix some factory_bot deprecations

As far as I can tell, wine_bouncer is fully compatible with grape 1.3. Nothing in the change log stands out as breaking and the tests seem compatible on all currently supported versions of grape, doorkeeper, and rails.
Additionally, I deployed wine_bouncer + grape 1.3 + rails 6 app and the API still functions.

I am open to feedback and collaboration on this as well 😃

### CODE REVIEW

**Comment by antek-drzewiecki** [comment]
Thanks for the PR! Ill take a closer look tonight and let you know :)

File: .travis.yml [hunk 1]
```diff
@@ -15,38 +15,119 @@ cache:
...
-    - rails=5.2.2 grape=1.2.3 doorkeeper=4.4.3
-    - rails=5.2.2 grape=1.0.0 doorkeeper=5.0.2
-    - rails=5.2.2 grape=1.1.0 doorkeeper=5.0.2
-    - rails=5.2.2 grape=1.2.3 doorkeeper=5.0.2
   global:
     CC_TEST_REPORTER_ID=ab1b6ce5f973da033f80ae2e99fadbb32b2f9c37892703956d8ef954c8e8134e
+  jobs:
+    - rails="~> 5.1.6.1" grape=0.19.2 doorkeeper=4.4.3
+    - rails="~> 5.1.6.1" grape=0.19.2 doorkeeper=5.0.2
+    - rails="~> 5.1.6.1" grape=0.19.2 doorkeeper=5.1.0
```
Comment (antek-drzewiecki) [inline .travis.yml]: Feel free to drop support for grape 0.19.2

**Comment by ghiculescu** [comment]
@sdfreund10 @antek-drzewiecki is it possible to get this released to rubygems? https://rubygems.org/gems/wine_bouncer still requires `grape < 1.3`

**Comment by rhomeister** [comment]
@sdfreund10 @antek-drzewiecki a release to rubygems would be highly appreciated! Thanks for your great work

**Comment by JustLime** [comment]
@sdfreund10 @antek-drzewiecki Still in the year 2022, a release to rubygems would be highly appreciated! Thanks for the great work!

**Pull request merged**